### ISSUE TITLE
saveFrame could not work properly with P3D

### ISSUE THREAD

**Issue by archgary** [processing/processing#4385]
This problem has been bothering me for a while. I've been using saveFrame for exporting images in 3D (P3D) since Processing 2.X. But the results of saveFrame with P3D in Processing keep giving me weird results (background changing, line overlapping...etc), like the attached images.

Is it because I am running in Windows 10 in high resolution? sound stupid, but this is really bothering me for a while. And I just keep going back to Processing 2.X for generating movies.
Thanks for anyone who answer the question in advance.

The code I tested is just a copy from official processing website of saveFrame function:
`int x = 0;
void draw() {
  background(204);
  if (x < 100) {
    line(x, 0, x, 100);
    x = x + 1;
  } else {
    noLoop();
  }
  // Saves each frame as line-000001.png, line-000002.png, etc.
  saveFrame("line-######.png");
}`

**Comment by archgary** [comment]
Find a solution with my Facebook friend, export in processing, Frederik Vanhoutte, pointed out it might be the problem with the graphic card. 

In my case, I didn't have separated graphic card with my Lenova Yoga 900, so I just need to tweak setting of my intel graphic card. 

There is a 3D option under my Intel HD Graphic Control Panel, and I just turn the "Conservative Morphological Anti-Aliasing" setting into Override Application Settings", then everything works fine then.

**Comment by archgary** [comment]
People who helped me also writing their solutions and I decided to put it here for people who has other machines. 

Basically, if you have 2 graphic cards, just force the better graphic card to run for java.exe.

1. Same problem on my Win10 portable. it has two video cards, an onboard Intel one and then a "real" NVidia. It is supposed to use the "right" card for each process. Processing seemed to use the crappy Intel. When I forced it to use the Nvidia the problem went away...

2. On the machine with two cards, go to Nvidia control panel and force it to use Nvidia card for java.exe and javaw.exe from processing folder (there is java runtime included).

Hope this helps most of the people with the saveFrame issue.

**Comment by JakubValtar** [comment]
@archgary Do you please have a screenshot of the glitch?

**Comment by archgary** [comment]
@JakubValtar I've attached them within the question post. Thank you.

**Comment by JakubValtar** [comment]
@archgary Thanks! Hm, there are cubes and the code is just drawing a line. If you have screenshots of the line (or code of the cubes), it would help a lot, because I can't compare it now.

**Comment by archgary** [comment]
@JakubValtar Sorry about my stupid mistakes. Now it should be clear, I didn't use PeasyCam as a 3D camera because I would like to keep it simple to figure out the problem. Now it should be clear enough. Thanks again.

**Comment by JakubValtar** [comment]
@archgary No worries, thanks! :) This is something we can work with.

**Comment by JakubValtar** [comment]
Any ideas @codeanticode? Driver problems, or are we doing something bad? (Apparently it works in 2 on the same machine.)

**Comment by benfry** [comment]
@JakubValtar is this #4578?

**Comment by JakubValtar** [comment]
@benfry No, this is unrelated. I updated to Windows 10 recently and I'm finally able to reproduce it. Let's see where it goes.

**Comment by JakubValtar** [comment]
I think Intel drivers can't do PBOs properly. The images I'm getting from the card don't have correct backgrounds (it's always default gray and black), it looks like primitives are accumulating there but clearing the background is ignored. When I check with OpenGL debugger, front and back buffers are ok, and I have no idea where are these images coming from.

@benfry I'm going to disable DMA transfers for Intel cards and just use blocking ReadPixels + AsyncImageSaver. It should be still pretty fast, DMA provides only a minor speedup.

**Comment by benfry** [comment]
Ok, I also disabled async `saveFrame()` by default in 3.1.2 in case you didn't catch that in the release notes.
